### PR TITLE
Refactors whip code to not break every single rule of programming

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request

title.
## Why It's Good For The Game
less shitcode good.

## Changelog
:cl:
refactor: whip isnt shitcode anymore
/:cl:

### CODE REVIEW

**Comment by TeomanTheGreat** [comment]
Good pr speedmerge immediately so I can fix my other pr

File: code/game/objects/items/melee/misc.dm [hunk 1]
```diff
@@ -677,55 +677,50 @@
...
+
+	switch(user.zone_selected)
+		if(BODY_ZONE_L_ARM)
+			whip_disarm(user, target, "left")
+		if(BODY_ZONE_R_ARM)
+			whip_disarm(user, target, "right")
+		if(BODY_ZONE_R_LEG)
+			whip_trip(user, target)
+		if(BODY_ZONE_L_LEG)
+			whip_trip(user, target)
```
Comment (Archanial) [inline code/game/objects/items/melee/misc.dm]: how are those 2 cases different?

File: code/game/objects/items/melee/misc.dm [hunk 2]
```diff
@@ -677,55 +677,50 @@
...
+		else
+			if(target.getarmor(type = "melee") < 16)
+				target.emote("scream")
+				target.visible_message("<span class='danger'>[user] whips [target]!</span>", "<span class='userdanger'>[user] whips you! It stings!</span>")
```
Comment (Archanial) [inline code/game/objects/items/melee/misc.dm]: shouldn't you always be checking for armor? Do you need that else here?

File: code/game/objects/items/melee/misc.dm (hunk 1 above)
Comment (Kapu1178) [reply code/game/objects/items/melee/misc.dm]: they arent. i dont know if BYOND switches let you do OR in conditions.

File: code/game/objects/items/melee/misc.dm (hunk 2 above)
Comment (Kapu1178) [reply code/game/objects/items/melee/misc.dm]: I assumed the intent was "if theyre wearing armor they wont feel the lashes" or some shit.

The else is to cover body parts that arent given an explicit condition in the switch. thats kind of an important thing.

File: code/game/objects/items/melee/misc.dm (hunk 1 above)
Comment (Kapu1178) [reply code/game/objects/items/melee/misc.dm]: (they do)

File: code/game/objects/items/melee/misc.dm [hunk 3]
```diff
@@ -677,55 +677,48 @@
...
-					if(!user.get_inactive_held_item())
-						U.throw_mode_on()
-						U.swap_hand()
-						I.throw_at(user, 10, 2)
+	if(!ishuman(target))
+		return
+
+	switch(user.zone_selected)
+		if(BODY_ZONE_L_ARM)
+			whip_disarm(user, target, "left")
```
Comment (PowerfulBacon) [inline code/game/objects/items/melee/misc.dm:693]: "left" and "right" should probably be a define

File: code/game/objects/items/melee/misc.dm [hunk 4]
```diff
@@ -677,55 +677,48 @@
...
+		if(BODY_ZONE_R_ARM)
+			whip_disarm(user, target, "right")
+		if(BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)
+			whip_trip(user, target)
+		else
+			if(target.getarmor(type = "melee") < 16)
+				target.emote("scream")
+				target.visible_message("<span class='danger'>[user] whips [target]!</span>", "<span class='userdanger'>[user] whips you! It stings!</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/melee/misc.dm]: This should also be another proc, proc/whip_harm or something

Comment (Kapu1178) [reply code/game/objects/items/melee/misc.dm]: done

File: code/game/objects/items/melee/misc.dm (hunk 3 above)
Comment (utilbird) [reply code/game/objects/items/melee/misc.dm:693]: Nah, the proc that arg is passed into uses strings. It'd be better if it were a define but that's probably beyond the scope of this PR

**Pull request merged**